docs/ccdproc/astrom_example.rst

```diff
@@ -0,0 +1,144 @@
+    from astropy.stats import mad_std
+    ccd = CCDData.read('sgpR.fits', unit='electron')
+    bkg_sigma = mad_std(ccd.data)
+    daofind = DAOStarFinder(fwhm=3.0, threshold=5.*std)
+    sources = daofind(ccd.data)
+    bright = sources[sources['mag'].argsort()[:20]]
+
+The current algorithm in photutils will result in duplicate objects in the list,
+especially for saturated stars or stars with diffraction spikes.   These should
+be removed from the list before proceeding.   A convenience function is provided
+in `ccdproc.astrometry`:
+
+.. code::
+
+    from ccdproc import astrometry as astrom
+    x, y = bright['xcentroid'], bright['ycentroid']
+    x, y = astrom.remove_duplicates(x, y, tolerance=10)
+
+This will remove all detections within 10 pixels of another detection, but leaving the first
+detection.
+
+Matching Coordinates
+--------------------
+
+Once the two coordinate lists have been produced, the next step is to match
+the two lists.   The function `ccdproc.astrometry.match_by_triangle` will
+produced a match list of coordinates.  The program first creates a series of
+triangles and looks for objects that have matching ratios to the two legs
+of the triangle created from the brightest object.  Further filtering can
+be done by matching that brightest triangle to a descending order of objects
+before doing a solution that matches all objects in the field.
+
+.. code::
+
+    r, d = gaia['RA_ICRS'], gaia['DE_ICRS']
+    matches = astrom.match_by_triangle(
+        x, y, r, d, n_limit=30, tolerance=0.02,
+        clean_limit=5, match_tolerance=0.02, m_init=models.Polynomial2D(1),
+        fitter = fitting.LinearLSQFitter())n_groups=5)
+    i1, i2 = matches
+
+This will produced a match catalog of stars that can be used to determine
+the transformation between the (x,y) -> (r,d).
+
+
+Deeper Matched Catalog
+----------------------
+
+Once an initial match is found, it is much easier to match a larger
+number of objects based on the initial transformation based on those
+coordinates.  The following command produces a list of indices that
+match between the two coordinate frames based on matched coordinates
+in the data set:
+
+.. code::
+
+    idp, idx = astrom.match_by_fit(
+        x, y, r.data, d.data, i1, i2, tolerance=5*u.arcsec)
+
+The match indices provide the full list of objects that match
+within the tolerance.  This can then be used to calculate the
+World Coordinate System (WCS) information for the image.
+
+Setting the WCS
+---------------
+
+Finally, from the set of matched frames and the image transformation, set
+the WCS for the image:
+
+.. code::
+
+    wcs = astrom.create_wcs_from_fit(x, y, r.data, d.data, idp, idw)
+    ccd.wcs = wcs
+
+The function calculates a WCS based on a linear transformation of the
+coordinates.  This does not currently include any higher distortion terms
+and so for the most accurate results, any distortion should be already
+removed from the image.
+
+
+References
+----------
+
+.. [1]
```

Comment:
I was a bit too lazy to fill in the actual links. But we shouldn't forget to actually include them before we merge it.